build/src/lib.rs

```diff
@@ -148,52 +281,59 @@ pub struct BinaryBuilder {
     fallback_to_build_dir: bool,
     build_dir: Option<PathBuf>,
     outputs: Vec<PathBuf>,
+    transform_path_fn: Option<Box<dyn TransformPathFn>>,
+    transform_data_fn: Option<Box<dyn TransformDataFn<Cursor<Vec<u8>>>>>,
 }
 
 impl BinaryBuilder {
+    /// ```plain
+    /// github release resource --[fn transform]--> $cache_dir --[fn postprocess]--> $OUT_DIR/
+    /// ```
+    ///
+    /// ```plain
+    /// github release resource --[fn transform]--> $OUT_DIR --[fn postprocess]--> $OUT_DIR/
+    /// ```
+    ///
     /// Acquires the rule and tokenizer binaries for one language by:
     /// - Trying to download them from their distribution source (or load them local cache).
     /// - If they are not found (i. e. a dev version of nlprule is used) and `fallback_to_build_dir`is true
     /// downloads the latest build directory  and builds the binaries from it.
     /// This can still fail if the dev version is sufficiently outdated for the latest build dir.
-    /// In that case, the user can't be helped.
-    fn build_language(&mut self, lang_code: &str) {
-        let tokenizer_out = self.out_dir.join(tokenizer_filename(lang_code));
-        let rules_out = self.out_dir.join(rules_filename(lang_code));
+    /// In that case, the user is encouraged to update to a release or a newer git sha.
+    fn build_language(&mut self, lang_code: &str) -> Result<()> {
+        // adjust the destination path for now
+        let path_transform = |out: PathBuf| -> Result<PathBuf> {
+            Ok(if let Some(ref fx) = self.transform_path_fn {
+                fx(out)?
+            } else {
+                out
+            })
+        };
+
+        let tokenizer_out = path_transform(self.out_dir.join(tokenizer_filename(lang_code)))?;
+        let rules_out = path_transform(self.out_dir.join(rules_filename(lang_code)))?;
 
         let mut did_not_find_binaries = false;
 
-        for (binary, out) in [
+        for (binary, out) in vec![
```

Comment:
Why `vec!` here? We can keep this as it is.

Reply:
Intention was to avoid the pointless copy in the first place until the impl of `IntoIterator` for arbitrary arrays `[]` (not slices!) stabilizes.